test/jobs/stock_purchase_job_test.rb

```diff
@@ -0,0 +1,21 @@
+require "test_helper"
+
+class StockPurchaseJobTest < ActiveJob::TestCase
+  test "it processes pending orders" do
+    pending_order1 = Order.create!(user: users(:one), stock: stocks(:one), status: "pending")
+    pending_order2 = Order.create!(user: users(:one), stock: stocks(:one), status: "pending")
+    completed_order = Order.create!(user: users(:one), stock: stocks(:one), status: "completed")
+
+    execute_calls = []
+
+    PurchaseStock.define_singleton_method(:execute) do |order|
```

Comment:
wasn't sure how to properly mock things out in minitest, this was what seemed to work?